### DISCUSSION TITLE
DMVPN Patches

### DISCUSSION THREAD

**Discussion by eskyuu** [strongswan/strongswan#610] [Ideas]
Hi,

I've just set up a DMVPN hub using FRR for routing and strongswan for IPSEC.  I came across 2 patches for strongswan that Alpine Linux are maintaining which allows the DMVPN hub to operate correctly.  From what I can see they essentially add another couple of events that can be registered through the VICI interface.

I am wondering if there has been any attempt to upstream these patches, or if there is a desire to get these integrated into the official strongswan repository.

The patches are located here:
https://gitlab.alpinelinux.org/alpine/aports/-/tree/master/main/strongswan

I have successfully built strongswan 5.9.1 after applying the 1002- and 1003- patches, and this has fixed the issues I was having with getting the DMVPN clients to connect correctly.

If it looks good, what is the best step that I can take to help get this change into the official repository?

**Comment by Thermi** [comment]
Oh look a wild PR appeared #502

**Reply by eskyuu** [reply]
Thanks for pointing me in the right direction.

Should any discussion go into the PR thread?  or are you suggesting that it's pointless trying to get this merged because it's already been covered?

I would be interested to understand how the GRE/NHRP/SS components all interact to make the DMVPN hub work, because I can see that the GRE config is not normal, and I suspect that the required interaction between these 3 components may have had some impact on why the SS patches were developed.
